az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopJavaJob.java

```diff
@@ -249,6 +249,22 @@ public String toString() {
         + getJobProps() + '}';
   }
 
+  /**
+   * Returns the {@code azkaban.execid} system property, if set, and otherwise returns the value of
+   * {@value CommonJobProperties#EXEC_ID}. {@value CommonJobProperties#EXEC_ID} may be overridden by a flow param or
+   * job property, whereas the {@code azkaban.execid} system property will be the unique execution id Azkaban generates
+   * for each flow.
+   */
+  private String getExecId() {
```

Comment:
I'm not sure if we want to piggyback on this tho, what if user intentionally wants to kill the exec_id he overrides with?

Reply:
User does not control the `cancel()` method that kills the job. The `cancel()` method should be able to find the log file that Azkaban created -- the log file is then searched for YARN application that need to be killed.